crates/ruma-events/src/do_not_disturb.rs

```diff
@@ -0,0 +1,152 @@
+//! Types for the [`dm.filament.do_not_disturb`] event.
+//!
+//! [`dm.filament.do_not_disturb`]: https://github.com/matrix-org/matrix-spec-proposals/pull/4359
+
+use std::collections::BTreeMap;
+
+use ruma_common::OwnedRoomId;
+use ruma_macros::EventContent;
+use serde::{Deserialize, Serialize};
+
+/// The content of a `dm.filament.do_not_disturb` event.
+///
+/// A list of rooms in "Do not Disturb" mode.
+#[derive(Clone, Debug, Default, Deserialize, Serialize, EventContent)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+#[ruma_event(type = "dm.filament.do_not_disturb", kind = GlobalAccountData)]
+pub struct DoNotDisturbEventContent {
+    /// A map of rooms in which to inhibit notifications.
+    ///
+    /// As [`DoNotDisturbRoom`] is currently empty, only the room IDs are useful and
+    /// can be accessed with the `.keys()` and `into_keys()` iterators.
+    pub rooms: BTreeMap<DoNotDisturbRoomIds, DoNotDisturbRoom>,
+}
+
+impl DoNotDisturbEventContent {
+    /// Creates a new `DoNotDisturbEventContent` from the given map of [`DoNotDisturbRoom`]s.
+    pub fn new(rooms: BTreeMap<DoNotDisturbRoomIds, DoNotDisturbRoom>) -> Self {
+        Self { rooms }
+    }
+
+    /// Creates a new `DoNotDisturbEventContent` from the given list of room IDs.
+    pub fn rooms(rooms: impl IntoIterator<Item = OwnedRoomId>) -> Self {
+        Self::new(
+            rooms
+                .into_iter()
+                .map(|id| (DoNotDisturbRoomIds::SingleRoom(id), DoNotDisturbRoom {}))
+                .collect(),
+        )
+    }
+}
+
+/// The key for a "Do not Disturb" setting.
+///
+/// This either matches a single room or all rooms.
+#[derive(Clone, Debug, Deserialize, Eq, Ord, PartialEq, PartialOrd, Serialize)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+pub enum DoNotDisturbRoomIds {
```

Comment:
I find it weird to have this end with an "s", given that it contains a single ID. Maybe it should be `DoNotDisturbRoomKey` or something?

Reply:
Yeah, good point. Renamed with 8dac0f0c8d2e30b0d9741f362de42ed66c44317c.